sdk/core/azure-core/inc/azure/core/internal/assert.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use with pre-conditions.
+ *
+ * @remark Asserts are turned ON when `NDEBUG` is NOT defined (for Debug build). For Release build,
+ * `std::abort()` is directly called if the condition is false, without calling assert().
+ *
+ */
+
+#pragma once
+
+#include "azure/core/platform.hpp"
+
+#include <cstdlib>
+#include <string>
+
+#if defined(NDEBUG)
+
+/*
+ * NDEBUG = defined = Build is on Release
+ * Define _azure_ASSERT to call abort directly on exp == false
+ */
+
+#define _azure_ASSERT(exp) \
```

Comment:
Other than the under-score renaming, putting the method in a namespace, and the file move/rename, any other changes to this file?

Doesn't look like it, but just verifying.

![image](https://user-images.githubusercontent.com/6527137/141416804-65feeb2c-7045-4e19-8211-e2ee6f37403b.png)

Reply:
No, only these two changes - prefix change and moving the function to a namespace.